enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,413 @@
+---
+title: Allowing version independence in siteconfig
+authors:
+  - "@sabbir-47"
+reviewers:
+  - "@imiller0"
+  - "@serngawy"
+  - "@Iack"
+  - "@serngawy" 
+approvers:
+  - "@imiller0"
+
+api-approvers:
+  - TBD
+creation-date: 2023-06-06
+last-updated: 2023-06-09
+status: provisional
+---
+
+
+# Cluster Group Upgrade (CGU) conditions and their reason/messages enhancement
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Documentation is updated in the upstream
+- [ ] Documentation is updated in the downstream
+
+
+## Summary
+
+Siteconfig allows customer to include custom extra manifests and list them in the siteconfig CR to be added during CR generation. It provides the flexibility to add additional CRs from the customer point of view during install time along with the CRs the RAN team provides. 
+
+The enhancement seeks to address the version independence of extra manifests, by putting version dependent manifests in version specific folder in git and pointing the directories via `extraManifestSearchPaths`.
+
+
+## Motivation
+
+* Currently customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs from git directory to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs. But it limits:
+
+    * does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+    * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+    * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
```

Comment:
I think we need to weave into the current behavior that the user is not able to capture the full set of version-specific extra manifests in git along with the SiteConfig because they cannot override the default set because of the inability to have the same named file.

Reply:
updated with commit https://github.com/openshift-kni/kni-enhancements/commit/dd384000bc29451c11ff6c6749e70cc07a1005d1